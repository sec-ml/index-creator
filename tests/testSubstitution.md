| term                                           | sub-term                                   | notes           | book | page |
| ---------------------------------------------- | ------------------------------------------ | --------------- | ---- | ---- |
| vuln^vulnerability                             |                                            | Vuln is bad.    | 1    | 10   |
| vuln                                           |                                            | vuln is common. |      |      |
| "we'll"^"we will"                              | we'll we'll rock you                       |                 |      |      |
| !we'll cancel this substitution out            |                                            |                 |      |      |
| !we'll escape\^escape this caret out           |                                            |                 |      |      |
| What about Vuln casing?                        | this test capitalisation of the first word |                 |      |      |
| WE'LL test capitalising all words              |                                            |                 |      |      |
| test maintaining acronyms as defined           | vms^"(VMs) virtual machines"               |                 |      |      |
| vms                                            | Vms                                        | VMS             |      |      |
| ttps^"tactics, techniques & procedures (TTPs)" | ttps, Ttps, TTPS                           |                 |      |      |
| gen^"general multi-word casing test"           | gen, Gen, GEN                              |                 |      |      |
