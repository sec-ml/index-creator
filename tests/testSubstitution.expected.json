[
  {
    "term": "(VMs) virtual machines",
    "subTerm": "(VMs) Virtual machines",
    "notes": "(VMs) Virtual Machines",
    "book": 1,
    "page": "10"
  },
  {
    "term": "general multi-word casing test",
    "subTerm": "general multi-word casing test, General multi-word casing test, General Multi-Word Casing Test",
    "notes": "",
    "book": 1,
    "page": "10"
  },
  {
    "term": "tactics, techniques & procedures (TTPs)",
    "subTerm": "tactics, techniques & procedures (TTPs), Tactics, techniques & procedures (TTPs), Tactics, Techniques & Procedures (TTPs)",
    "notes": "",
    "book": 1,
    "page": "10"
  },
  {
    "term": "test maintaining acronyms as defined",
    "subTerm": "(VMs) virtual machines",
    "notes": "",
    "book": 1,
    "page": "10"
  },
  {
    "term": "vulnerability",
    "subTerm": "",
    "notes": "Vulnerability is bad.",
    "book": 1,
    "page": "10"
  },
  {
    "term": "vulnerability",
    "subTerm": "",
    "notes": "vulnerability is common.",
    "book": 1,
    "page": "10"
  },
  {
    "term": "we will",
    "subTerm": "we will we will rock you",
    "notes": "",
    "book": 1,
    "page": "10"
  },
  {
    "term": "We Will test capitalising all words",
    "subTerm": "",
    "notes": "",
    "book": 1,
    "page": "10"
  },
  {
    "term": "we'll cancel this substitution out",
    "subTerm": "",
    "notes": "",
    "book": 1,
    "page": "10"
  },
  {
    "term": "we'll escape\\^escape this caret out",
    "subTerm": "",
    "notes": "",
    "book": 1,
    "page": "10"
  },
  {
    "term": "What about Vulnerability casing?",
    "subTerm": "this test capitalisation of the first word",
    "notes": "",
    "book": 1,
    "page": "10"
  }
]
